### PR TITLE
Add AI analysis prompt generator to config exporter

### PR DESCRIPTION
**Pull request by justin808**
## Summary

Adds automatic AI analysis prompt generation to the `bin/export-bundler-config --doctor` command. This creates an `AI-ANALYSIS-PROMPT.md` file alongside exported configurations, providing comprehensive instructions for AI assistants to analyze webpack/rspack configs.

## Key Improvements

- **AI-Powered Configuration Analysis**: Users can now get intelligent feedback on their bundler configurations by copying the prompt + exported configs to ChatGPT, Claude, Gemini, etc.
- **Comprehensive Analysis Framework**: The prompt guides AI to check for migration issues, build errors, optimization opportunities, and best practices
- **React on Rails Context**: Includes specific guidance for React on Rails standard configuration patterns (client/server bundles, SSR setup, etc.)

## What's Included in the AI Prompt

The generated prompt instructs AI to analyze:

1. **Migration Issues**: webpack ↔ rspack compatibility, deprecated options, equivalent replacements
2. **Build Errors & Warnings**: Configuration problems that cause build failures
3. **Client vs Server Optimization**: Bundle-specific optimizations for browser vs SSR
4. **Development vs Production**: Environment-specific settings (HMR, minification, source maps, etc.)
5. **Best Practices**: Caching, loaders, plugins, module resolution

## Example Output

When running `bin/export-bundler-config --doctor`, you now get:

```
Configuration Files:
  ✓ webpack-development-client.yaml
  ✓ webpack-development-server.yaml
  ✓ webpack-production-client.yaml
  ✓ webpack-production-server.yaml

AI Analysis:
  🤖 AI-ANALYSIS-PROMPT.md - Use this prompt to get AI recommendations
     Copy the contents and paste into an AI assistant for configuration analysis
```

## Test Plan

- [x] All existing tests pass
- [x] New comprehensive test suite for `AiPromptGenerator` (11 tests)
- [x] Linter passes
- [x] TypeScript type check passes
- [x] Verified prompt generation in doctor mode
- [x] Updated troubleshooting documentation

## Breaking Changes

None - this is purely additive functionality.

## Security Implications

None - generates documentation files only, no code execution or external dependencies.

## Summary by CodeRabbit

* **New Features**
  * Doctor mode now generates an AI analysis prompt file alongside configuration exports, providing guided analysis for migration issues, build errors, optimization opportunities, and best practices.

* **Documentation**
  * Updated troubleshooting guide to document AI prompt generation in doctor mode and how to use exported artifacts with AI analysis tools.

---

> [!NOTE]
> **Low Risk**
> Additive doctor-mode documentation output only; failures are non-fatal warnings and existing export behavior is preserved aside from the intentional HMR duplicate-write fix.
> 
> **Overview**
> **Doctor mode** (`bin/shakapacker-config --doctor`) now writes a companion **`AI-ANALYSIS-PROMPT.md`** next to exported configs. The new `AiPromptGenerator` builds a paste-ready markdown prompt (bundler context, labeled export filenames, React on Rails notes, and analysis objectives for migration, builds, client/server, dev/prod, and best practices). Export directory context uses only the **basename** of the target path to avoid leaking absolute paths.
> 
> `writeAiAnalysisPrompt` runs after exports in both config-file and fallback doctor flows: it deduplicates config basenames, supports mixed webpack/rspack labels, **warns** if the prompt file cannot be written, and leaves exported YAML/JSON unchanged. The doctor summary lists config files and the AI prompt separately.
> 
> A related **doctor fix** limits the development **HMR** pass to **client** (and `all`) configs so server bundles are not written twice under the same name. **CHANGELOG** and **troubleshooting** docs describe the new artifact and expected doctor filenames (including HMR client).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit bab00538c22c42184c0c83c33af8a680fb4ed4e2. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by justin808** [comment]
Closing as stale: this PR is large, has long-standing requested changes, and is blocking on substantial rework. Re-opening as smaller focused PRs will be easier to land and review.

File: package/configExporter/cli.ts [hunk 1]
```diff
@@ -1022,24 +1033,88 @@ async function runDoctorMode(
...
+    return null
+  }
+  try {
+    const aiPromptGenerator = new AiPromptGenerator()
+    const bundler = [...bundlers].join(" and ") || "unknown"
+    const fileBasenames = createdFiles.map((f) => basename(f))
+    const aiPromptContent = aiPromptGenerator.generatePrompt(
+      fileBasenames,
+      targetDir,
+      bundler
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts:1088]: Fixed in 9e3713d2: the HMR pass now skips non-client configs so the dev server bundle is written once instead of by both the HMR and regular development passes, and `writeAiAnalysisPrompt` dedups basenames as defense-in-depth. Added a regression test asserting each config is listed once.

File: package/configExporter/aiPromptGenerator.ts [hunk 2]
```diff
@@ -0,0 +1,330 @@
+import { basename } from "path"
+import { ConfigMetadata } from "./types"
+
+/**
+ * Generates an AI prompt file for analyzing webpack/rspack configuration exports.
+ * The prompt helps AI identify issues and suggest optimizations for the bundler configuration.
+ */
+export class AiPromptGenerator {
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts:9]: Fixed in 9e3713d2: `generatePrompt` and `generatePromptFilename` are now `static`, so callers no longer need `new AiPromptGenerator()` (matching the existing `FileWriter` pattern).

File: test/package/configExporter/aiPromptGenerator.test.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    const targetDir = "/path/to/exports"
+    const bundler = "webpack"
+
+    const prompt = generator.generatePrompt(exportedFiles, targetDir, bundler)
+
+    expect(prompt).toContain("Generated by Shakapacker Config Exporter")
+    expect(prompt).toContain("**Bundler**: webpack")
+    expect(prompt).toMatch(/\*\*Date\*\*: \d{4}-\d{2}-\d{2}T/)
+  })
+})
```
[the remark this answers is not in the document]
Comment (justin808) [reply test/package/configExporter/aiPromptGenerator.test.js:352]: Already addressed: the test file ends with a trailing newline in the current revision.

File: package/configExporter/aiPromptGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,299 @@
+import { basename } from "path"
+import { ConfigMetadata } from "./types"
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed in the current revision: `aiPromptGenerator.ts` no longer imports `basename` or `ConfigMetadata` — the file has no imports.

File: package/configExporter/aiPromptGenerator.ts [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed in the current revision: `aiPromptGenerator.ts` no longer imports `basename` or `ConfigMetadata` — the file has no imports.

File: package/configExporter/cli.ts [hunk 6]
```diff
@@ -845,8 +848,21 @@ async function runDoctorMode(
...
+        )
+        const aiPromptFilename = aiPromptGenerator.generatePromptFilename()
+        const aiPromptPath = resolve(targetDir, aiPromptFilename)
+        FileWriter.writeSingleFile(aiPromptPath, aiPromptContent)
+        createdFiles.push(aiPromptPath)
+
         // Print summary and exit early
-        printDoctorSummary(createdFiles, targetDir)
+        printDoctorSummary(createdFiles, targetDir, aiPromptFilename)
         return
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Already addressed: the duplicated block has been extracted into a single shared `writeAiAnalysisPrompt` helper that both doctor-mode branches call.

File: package/configExporter/aiPromptGenerator.ts [hunk 7]
```diff
@@ -0,0 +1,330 @@
+import { basename } from "path"
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts:1]: Already addressed in the current revision: `aiPromptGenerator.ts` no longer imports `basename` or `ConfigMetadata` — the file has no imports.

File: package/configExporter/aiPromptGenerator.ts [hunk 8]
```diff
@@ -0,0 +1,330 @@
...
+    sections.push(
+      "Please analyze these configurations and provide recommendations."
+    )
+    sections.push("")
+
+    // Context section
+    sections.push("## Context")
+    sections.push("")
+    sections.push(`- **Bundler**: ${bundler}`)
+    sections.push(`- **Exported At**: ${new Date().toISOString()}`)
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed: the timestamp is captured once (`const timestamp = new Date().toISOString()`) and reused in both the Context section and the footer.

File: package/configExporter/aiPromptGenerator.ts [hunk 9]
```diff
@@ -0,0 +1,330 @@
...
+      sections.push("- Production client")
+      sections.push("- Production server (SSR)")
+      sections.push("")
+      sections.push(
+        "The HMR configs are especially useful for troubleshooting dev server, hot reload,"
+      )
+      sections.push("and live reload issues.")
+      sections.push("")
+      sections.push(
+        "Track progress at: https://github.com/shakacode/react_on_rails/issues/1862"
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed: the "Coming Soon" / issue-1862 block has been removed; the prompt now links to the stable React on Rails example configs on `main`.

File: package/configExporter/cli.ts [hunk 10]
```diff
@@ -1022,24 +1039,54 @@ async function runDoctorMode(
...
   console.log("=".repeat(80))
-  console.log(`\nCreated ${createdFiles.length} configuration file(s) in:`)
+  console.log(`\nCreated ${createdFiles.length} file(s) in:`)
   console.log(`  ${targetDir}\n`)
-  console.log("Files:")
+  console.log("Configuration Files:")
   createdFiles.forEach((file) => {
-    console.log(`  ✓ ${basename(file)}`)
+    const name = basename(file)
+    if (name.endsWith(".md")) {
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Already addressed: the summary no longer filters on the `.md` extension — it compares against the exact prompt filename (`name === aiPromptFilename`).

File: package/configExporter/aiPromptGenerator.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed in the current revision: `aiPromptGenerator.ts` no longer imports `basename` or `ConfigMetadata` — the file has no imports.

File: package/configExporter/aiPromptGenerator.ts [hunk 11]
```diff
@@ -0,0 +1,330 @@
...
+    sections.push("")
+    sections.push(`- **Bundler**: ${bundler}`)
+    sections.push(`- **Exported At**: ${new Date().toISOString()}`)
+    sections.push(`- **Export Directory**: ${targetDir}`)
+    sections.push("")
+
+    // Configuration files
+    sections.push("## Configuration Files")
+    sections.push("")
+    sections.push(
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed: the timestamp is captured once (`const timestamp = new Date().toISOString()`) and reused in both the Context section and the footer.

File: package/configExporter/aiPromptGenerator.ts [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+      sections.push("")
+    }
+
+    // Analysis requests
+    sections.push("## Analysis Objectives")
+    sections.push("")
+    sections.push(
+      "Please analyze the exported configuration files and provide:"
+    )
+    sections.push("")
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed: the "Coming Soon" / issue-1862 block has been removed; the prompt now links to the stable React on Rails example configs on `main`.

File: package/configExporter/cli.ts [hunk 13]
```diff
@@ -932,8 +935,21 @@ async function runDoctorMode(
...
+          detectedBundler
+        )
+        const aiPromptFilename = aiPromptGenerator.generatePromptFilename()
+        const aiPromptPath = resolve(targetDir, aiPromptFilename)
+        FileWriter.writeSingleFile(aiPromptPath, aiPromptContent)
+        createdFiles.push(aiPromptPath)
+
         // Print summary and exit early
-        printDoctorSummary(createdFiles, targetDir)
+        printDoctorSummary(createdFiles, targetDir, aiPromptFilename)
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Already addressed: the duplicated block has been extracted into a single shared `writeAiAnalysisPrompt` helper that both doctor-mode branches call.

File: package/configExporter/cli.ts [hunk 14]
```diff
@@ -887,6 +888,7 @@ async function runDoctorMode(
     const targetDir = options.saveDir! // Set by applyDefaults
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts:889]: Already addressed: doctor mode tracks every bundler in a `detectedBundlers` Set and labels the prompt with the joined value, falling back to "unknown" when empty, so an empty config loop no longer mislabels the bundler.

**Comment by justin808** [comment]
## Review follow-up (commit 9e3713d2)

Worked through the outstanding review feedback. Most of the earlier comments were already resolved by prior commits on this branch; the remaining substantive items are now addressed.

**Fixed in this commit:**

- **Duplicate configs in the AI prompt** (@cursor[bot]): In default `--doctor` mode the development server bundle was written twice — once by the HMR pass and once by the regular development pass — under the same path, so the file count, the printed Configuration Files list, and `AI-ANALYSIS-PROMPT.md` each listed it twice. The HMR pass now skips non-client configs (HMR only applies to client bundles), and `writeAiAnalysisPrompt` deduplicates basenames as defense-in-depth. Added a regression test.
- **Stateless generator** (@claude[bot]): `AiPromptGenerator.generatePrompt` and `generatePromptFilename` are now `static`, so callers no longer need `new AiPromptGenerator()` — matching the existing `FileWriter` pattern.
- **Edge-case test** (@claude[bot]): Strengthened the empty-`exportedFiles` test to assert the prompt still renders a valid structure (header, context, objectives, footer).

`yarn jest test/package/configExporter/` (102 tests), `tsc --noEmit`, and `eslint` all pass.

**Already addressed in earlier commits** (these threads have been resolved with per-thread notes): unused `basename`/`ConfigMetadata` imports, the duplicated prompt-generation block (extracted into `writeAiAnalysisPrompt`), the double `new Date()` (now a single reused `timestamp`), the stale "Coming Soon"/issue-1862 block (removed), the fragile `.md` filter (now an exact filename match), the `detectedBundler` default (now a `detectedBundlers` Set with an `"unknown"` fallback), missing trailing newlines, and the `index.d.ts` rewrite (reverted in 855fafbe).

**Left open for a maintainer decision** (two threads intentionally unresolved):

1. **Absolute `targetDir` embedded in the shareable prompt** (@greptile-apps[bot]): The full local export path is written into a file users are told to paste into external AI assistants, which can leak a username / directory layout. Options: keep as-is (useful context), strip to `basename(targetDir)`, or omit.
2. **`includeReactOnRailsContext` defaults to `true` and is never overridden** (@claude[bot]): Every Shakapacker user gets the React on Rails section. Options: auto-detect `react_on_rails`, default to `false`, or keep as-is.

Happy to take either direction on (1) and (2).

File: package/configExporter/aiPromptGenerator.ts [hunk 15]
```diff
@@ -0,0 +1,330 @@
...
+      "Please analyze these configurations and provide recommendations."
+    )
+    sections.push("")
+
+    // Context section
+    sections.push("## Context")
+    sections.push("")
+    sections.push(`- **Bundler**: ${bundler}`)
+    sections.push(`- **Exported At**: ${new Date().toISOString()}`)
+    sections.push(`- **Export Directory**: ${targetDir}`)
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Fixed in 22181e0d: the prompt now shows only `basename(targetDir)` under **Export Directory**, so the absolute local path (and any username it embeds) is no longer written into a file meant to be pasted into external AI assistants. Added a test asserting the base name is present and the full path is not.

File: package/configExporter/aiPromptGenerator.ts [hunk 16]
```diff
@@ -0,0 +1,330 @@
...
+   * @param targetDir - Directory where configs were exported
+   * @param bundler - The bundler type (webpack or rspack)
+   * @param includeReactOnRailsContext - Whether to include React on Rails specific context
+   * @returns Markdown-formatted AI prompt
+   */
+  generatePrompt(
+    exportedFiles: string[],
+    targetDir: string,
+    bundler: string,
+    includeReactOnRailsContext = true
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Good call. Deferring this to a focused follow-up to keep #695 small (it was previously closed for being too large). Tracked in #1162, which proposes auto-detecting `react_on_rails` (npm package and/or Ruby gem) and driving `includeReactOnRailsContext` from that instead of defaulting it on. Resolving here; discussion continues on the issue.

File: package/configExporter/aiPromptGenerator.ts [hunk 17]
```diff
@@ -0,0 +1,326 @@
...
+    } else if (lower.includes("test")) {
+      mode = "test"
+    }
+    const modeLabel = mode ? `${mode[0].toUpperCase()}${mode.slice(1)}` : null
+    const modeSuffix = mode ? ` for ${mode} mode` : ""
+
+    if (lower.includes("hmr")) {
+      const label = modeLabel ? `${modeLabel} (Client, HMR)` : "Client (HMR)"
+      return `- **${label}**: \`${filename}\` - Client bundle with Hot Module Replacement${modeSuffix}`
+    }
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts:257]: Fixed in 3884c2c8: `describeExportedFile` now checks for `server` before `hmr`/`client`, so a combined build name like `webpack-dev-hmr-server.yml` is labeled as a server bundle. Added a test.

File: package/configExporter/aiPromptGenerator.ts [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+   * @param targetDir - Directory where configs were exported
+   * @param bundler - The bundler label (e.g. "webpack", "rspack", or "webpack and rspack" for mixed-bundler exports)
+   * @param includeReactOnRailsContext - Whether to include React on Rails specific context (defaults to true)
+   * @returns Markdown-formatted AI prompt
+   */
+  static generatePrompt(
+    exportedFiles: string[],
+    targetDir: string,
+    bundler: string,
+    includeReactOnRailsContext = true
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Duplicate of the thread above, which is tracked as a focused follow-up in #1162 (auto-detect `react_on_rails` and drive `includeReactOnRailsContext` from that, rather than adding a CLI flag). Resolving here; discussion continues on the issue.

File: package/configExporter/cli.ts [hunk 19]
```diff
@@ -1022,24 +1043,90 @@ async function runDoctorMode(
...
+    const fileBasenames = [...new Set(createdFiles.map((f) => basename(f)))]
+    const aiPromptContent = AiPromptGenerator.generatePrompt(
+      fileBasenames,
+      targetDir,
+      bundler
+    )
+    const aiPromptFilename = AiPromptGenerator.generatePromptFilename()
+    const aiPromptPath = resolve(targetDir, aiPromptFilename)
+    FileWriter.writeSingleFile(aiPromptPath, aiPromptContent)
+    createdFiles.push(aiPromptPath)
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Leaving as-is: the mutate-then-return pattern is consistent with the rest of `runDoctorMode`, which appends to `createdFiles` throughout the export loops. Keeping `writeAiAnalysisPrompt` aligned with that convention avoids a one-off signature.

File: package/configExporter/cli.ts [hunk 20]
```diff
@@ -1022,24 +1043,90 @@ async function runDoctorMode(
...
+  createdFiles: string[],
+  targetDir: string,
+  aiPromptFilename: string | null
+): void {
   // Print summary
   console.log(`\n${"=".repeat(80)}`)
   console.log("✅ Export Complete!")
   console.log("=".repeat(80))
-  console.log(`\nCreated ${createdFiles.length} configuration file(s) in:`)
+  console.log(`\nCreated ${createdFiles.length} file(s) in:`)
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Not a bug: the AI prompt is counted and shown — under the separate `AI Analysis:` line, not the `Configuration Files:` list. So `Created 5 file(s)` is 4 config files plus the 1 prompt file, all displayed. The count matches what is written to disk, so no change here.

File: package/configExporter/aiPromptGenerator.ts [hunk 21]
```diff
@@ -0,0 +1,326 @@
...
+      const label = modeLabel ? `${modeLabel} (All bundles)` : "All bundles"
+      return `- **${label}**: \`${filename}\` - Combined client and server configuration${modeSuffix}`
+    }
+    return `- \`${filename}\``
+  }
+
+  /**
+   * Generate the filename for the AI prompt file.
+   */
+  static generatePromptFilename(): string {
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Leaving as a method: `generatePromptFilename()` is the pre-existing public API on `AiPromptGenerator` (used by the CLI and tests), so keeping it a method avoids churning that surface for a cosmetic change. Happy to revisit if we adopt a constants convention for the exporter.

File: package/configExporter/aiPromptGenerator.ts [hunk 22]
```diff
@@ -0,0 +1,326 @@
...
+      sections.push("### Reference Configuration Examples")
+      sections.push("")
+      sections.push(
+        "For comparison, you can reference the standard React on Rails configuration examples:"
+      )
+      sections.push("")
+      sections.push("**Source Configuration Files (JavaScript):**")
+      sections.push("")
+      sections.push(
+        "- [commonWebpackConfig.js](https://github.com/shakacode/react_on_rails/blob/main/react_on_rails/spec/dummy/config/webpack/commonWebpackConfig.js)"
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Good point on durability. This sits inside the React on Rails reference section that #1162 will rework, so I have noted the link-stability concern there (prefer durable docs over `spec/dummy` fixture paths). Resolving here and tracking it on the issue.

File: package/configExporter/aiPromptGenerator.ts [hunk 23]
```diff
@@ -0,0 +1,334 @@
...
+   * @param targetDir - Directory where configs were exported (only its base name appears in the prompt)
+   * @param bundler - The bundler label (e.g. "webpack", "rspack", or "webpack and rspack" for mixed-bundler exports)
+   * @param includeReactOnRailsContext - Whether to include React on Rails specific context (defaults to true)
+   * @returns Markdown-formatted AI prompt
+   */
+  static generatePrompt(
+    exportedFiles: string[],
+    targetDir: string,
+    bundler: string,
+    includeReactOnRailsContext = true
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Tracked in #1162: gating the React on Rails section on `react_on_rails` detection (rather than the always-`true` default) is deferred to a focused follow-up so #695 stays small. Resolving here.

File: package/configExporter/aiPromptGenerator.ts [hunk 24]
```diff
@@ -0,0 +1,334 @@
...
+        "- [commonWebpackConfig.js](https://github.com/shakacode/react_on_rails/blob/main/react_on_rails/spec/dummy/config/webpack/commonWebpackConfig.js)"
+      )
+      sections.push(
+        "- [clientWebpackConfig.js](https://github.com/shakacode/react_on_rails/blob/main/react_on_rails/spec/dummy/config/webpack/clientWebpackConfig.js)"
+      )
+      sections.push(
+        "- [serverWebpackConfig.js](https://github.com/shakacode/react_on_rails/blob/main/react_on_rails/spec/dummy/config/webpack/serverWebpackConfig.js)"
+      )
+      sections.push(
+        "- [webpack.config.js](https://github.com/shakacode/react_on_rails/blob/main/react_on_rails/spec/dummy/config/webpack/webpack.config.js)"
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Noted on #1162, which reworks this React on Rails reference section and calls out swapping the `spec/dummy` fixture links for durable documentation URLs. Resolving here; tracked on the issue.

File: package/configExporter/aiPromptGenerator.ts [hunk 25]
```diff
@@ -0,0 +1,334 @@
...
+      return `- **${label}**: \`${filename}\` - Client bundle${modeSuffix}`
+    }
+    if (lower.includes("-all.")) {
+      const label = modeLabel ? `${modeLabel} (All bundles)` : "All bundles"
+      return `- **${label}**: \`${filename}\` - Combined client and server configuration${modeSuffix}`
+    }
+    return `- \`${filename}\``
+  }
+
+  /**
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Leaving `generatePromptFilename()` as a method: it is the pre-existing public API on `AiPromptGenerator` (used by the CLI and tests), so keeping it avoids churning that surface for a cosmetic change. Same call as the sibling thread on this line.

File: package/configExporter/cli.ts (hunk 19 above)
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Leaving as-is: the mutate-then-return pattern matches the rest of `runDoctorMode`, which appends to `createdFiles` throughout the export loops, and the summary count is correct (the prompt is shown under the separate `AI Analysis:` line). Same call as the sibling thread on this function.

File: package/configExporter/aiPromptGenerator.ts [hunk 26]
```diff
@@ -0,0 +1,334 @@
...
+    sections.push("- Performance budgets")
+    sections.push("")
+
+    sections.push("### 5. Common Best Practices")
+    sections.push("")
+    sections.push("Check for adherence to bundler best practices:")
+    sections.push("")
+    sections.push("- Efficient caching strategies")
+    sections.push("- Proper externals configuration")
+    sections.push("- Loader performance (e.g., thread-loader, cache-loader)")
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Done in bd6f534 — replaced the `cache-loader` recommendation with webpack 5's built-in persistent caching (`cache: { type: 'filesystem' }`).

File: package/configExporter/cli.ts (hunk 19 above)
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts]: Done in bd6f534 — `writeAiAnalysisPrompt` no longer mutates `createdFiles`. It now only writes the file and returns the filename; the caller passes that to `printDoctorSummary`, which reports the prompt separately. `createdFiles` stays a pure record of exported configs.

File: package/configExporter/aiPromptGenerator.ts [hunk 27]
```diff
@@ -0,0 +1,334 @@
...
+   * @returns Markdown-formatted AI prompt
+   */
+  static generatePrompt(
+    exportedFiles: string[],
+    targetDir: string,
+    bundler: string,
+    includeReactOnRailsContext = true
+  ): string {
+    const timestamp = new Date().toISOString()
+    const sections: string[] = []
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Done in bd6f534 — rebuilt `generatePrompt` from the ~170 `sections.push()` calls into a single template literal, with two small helpers for the dynamic config-files body and the static React on Rails block. The document structure is now legible at a glance.

**Comment by justin808** [comment]
## Review follow-up (commit bd6f534)

Addressed the remaining actionable review points from the latest code reviews:

- **Deprecated `cache-loader` hint** → replaced with webpack 5's built-in persistent caching (`cache: { type: 'filesystem' }`).
- **`writeAiAnalysisPrompt` array mutation** → it no longer pushes into `createdFiles`; it just returns the filename and the doctor summary reports the prompt separately.
- **`sections.push()` template** → `generatePrompt` is now a single template literal with two small helpers (config-files body + static RoR block).
- **`generatePromptFilename()`** → replaced with a `static readonly PROMPT_FILENAME` constant.
- **Reference links to `spec/dummy` fixtures** → now point at durable destinations: the [Webpack Configuration docs guide](https://reactonrails.com/docs/core-concepts/webpack-configuration) and the example app's webpack config directory (one directory link instead of four individual fixture-file links).
- **Doctor summary file count** → now spelled out explicitly ("N configuration file(s) and 1 AI analysis prompt") so it matches the lists below it.

Tests updated; full JS suite (548) and `yarn lint` pass.

Two items intentionally left open for a maintainer decision:
1. **`includeReactOnRailsContext` default** — keep `true`, default to `false`, auto-detect `react_on_rails` in package.json, or expose a CLI flag.
2. **Extracting the HMR server-bundle dedup fix** into its own PR.

File: package/configExporter/aiPromptGenerator.ts (hunk 23 above)
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Done in 5f78d46 — went with option 1: removed the `includeReactOnRailsContext` parameter and the section is now always included. It was never set from any production caller, so per the project's open-source maintainability rules it was pure surface. The block stays gated behind "If this project uses React on Rails…", so non-RoR users (and any AI reading the prompt) simply skip it.

File: package/configExporter/cli.ts [hunk 28]
```diff
@@ -981,7 +991,18 @@ async function runDoctorMode(
       // eslint-disable-next-line no-await-in-loop -- Sequential execution required: each config modifies shared global state (env vars, config cache) that must be cleared/restored between iterations
       const configs = await loadConfigsForEnv(env, options, appRoot)
 
-      for (const { config, metadata } of configs) {
+      // HMR only applies to client bundles. During the HMR pass, skip
+      // server/other configs so they are written once (by the regular
+      // development pass) instead of being duplicated under the same filename.
+      const configsToWrite = hmr
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts:997]: Decision: keeping this in the current PR rather than extracting it. The dedup fix is scoped to the same area as the feature (doctor-mode export), it's small and has its own test, and #695 was already closed-as-stale once — a separate-PR round-trip is friction that outweighs the purity benefit here. Resolving as won't-extract.

File: package/configExporter/aiPromptGenerator.ts [hunk 29]
```diff
@@ -0,0 +1,274 @@
...
+   * @param targetDir - Directory where configs were exported (only its base name appears in the prompt)
+   * @param bundler - The bundler label (e.g. "webpack", "rspack", or "webpack and rspack" for mixed-bundler exports)
+   * @param includeReactOnRailsContext - Whether to include React on Rails specific context (defaults to true)
+   * @returns Markdown-formatted AI prompt
+   */
+  static generatePrompt(
+    exportedFiles: string[],
+    targetDir: string,
+    bundler: string,
+    includeReactOnRailsContext = true
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Already addressed in 5f78d46, which landed after this review pass was generated: the `includeReactOnRailsContext` parameter was removed entirely and the section is now always included (gated by the "If this project uses React on Rails…" phrasing). Resolving.

File: package/configExporter/cli.ts (hunk 28 above)
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts:997]: Acknowledged — by maintainer decision this stays in the current PR rather than being extracted. The dedup is scoped to doctor-mode export (same area as the feature), is small and has its own test, and #695 was closed-as-stale once, so a separate-PR round-trip isn't worth the friction. Resolving as won't-extract.

File: package/configExporter/aiPromptGenerator.ts [hunk 30]
```diff
@@ -0,0 +1,274 @@
...
+   - Uses \`css-loader\` with \`exportOnlyLocals: true\`
+   - Target can be 'web' or 'node' depending on renderer
+
+### Reference Configuration Examples
+
+For comparison, see the React on Rails webpack/rspack documentation and the
+maintained example app configuration:
+
+- [Webpack Configuration guide](https://reactonrails.com/docs/core-concepts/webpack-configuration)
+- [Example app webpack configs](https://github.com/shakacode/react_on_rails/tree/main/react_on_rails/spec/dummy/config/webpack)
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts]: Done in bab00538 — dropped the `spec/dummy` example-app link and kept only the canonical [Webpack Configuration docs guide](https://reactonrails.com/docs/core-concepts/webpack-configuration), which is the durable public reference. No fixture paths remain in the generated prompt.

File: package/configExporter/cli.ts [hunk 31]
```diff
@@ -1022,24 +1043,95 @@ async function runDoctorMode(
...
+/**
+ * Generate and write the AI analysis prompt that accompanies a doctor-mode
+ * export. The prompt is a best-effort companion file: the exported configs are
+ * the primary deliverable, so failures here warn instead of aborting the
+ * export. Returns the prompt filename, or null if nothing was written.
+ *
+ * `createdFiles` is read (to derive the bundler list and config names) but not
+ * mutated; the caller owns the prompt path via the return value.
+ *
+ * Exported for testing.
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/cli.ts:1069]: Leaving this as-is. `cli.ts` is the CLI entrypoint, not the package's public API surface (that's `index.ts`, which does not re-export this helper), so the breaking-change concern is hypothetical. The "export for testing" pattern is already used elsewhere in this file (e.g. `parseArguments`), and spinning up a dedicated `aiPromptWriter.ts` module for a single best-effort helper adds structure that isn't warranted for this feature. If we later grow a public exporter API, extracting it then would be the right time. Resolving as won't-do.

File: package/configExporter/aiPromptGenerator.ts [hunk 32]
```diff
@@ -0,0 +1,269 @@
...
+
+    return `The following configuration files are available for analysis:\n\n${fileList}${hmrNote}`
+  }
+
+  /**
+   * Static React on Rails reference block. Ends with a trailing blank line so it
+   * can be interpolated directly before the next section without disturbing the
+   * surrounding spacing.
+   */
+  private static readonly REACT_ON_RAILS_CONTEXT = `## React on Rails Standard Configuration
```
[the remark this answers is not in the document]
Comment (justin808) [reply package/configExporter/aiPromptGenerator.ts:191]: This is a deliberate, already-made decision (see the resolved thread on the same parameter): the React on Rails section is intentionally always included because Shakapacker is primarily used with React on Rails, and the "If this project uses React on Rails…" gating makes it skippable for everyone else. We removed the toggle parameter rather than add configuration surface, per the project's open-source maintainability rules. Resolving as by-design.
